### PR TITLE
feat(chore): dev-compactor

### PR DESCRIPTION
**Pull request by CAJan93**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Needed to test the virtual compactor locally https://github.com/risingwavelabs/risingwave-extensions/pull/36/

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: risedev.yml [hunk 1]
```diff
@@ -92,6 +92,17 @@ profile:
...
+  # You can use this in combination with the virtual compactor
+  # provided in https://github.com/risingwavelabs/risingwave-extensions
+  dev-compactor:
+    steps:
+      - use: minio
+      - use: meta-node
+      - use: compute-node
+      - use: frontend
+      - use: compactor
+        user-managed: true
```
Comment (yufansong) [inline risedev.yml:104]: Little confuse about why this can used for dev compactor.

**Review by arkbriar — APPROVED** [review]
LGTM!

File: risedev.yml (hunk 1 above)
Comment (CAJan93) [reply risedev.yml:104]: `user-managed` just means that meta and CN expects there to be a compactor, but that this compactor will be started by the user. You could start your compactor in debug mode OR in our case you start a virtual compactor with the params defined in https://github.com/risingwavelabs/risingwave-extensions/pull/36.

Comment (yufansong) [reply risedev.yml:104]: Get

**Review by yufansong — APPROVED** [review]

**Pull request merged**